src/main/java/Sprites.java

```diff
@@ -0,0 +1,112 @@
+import org.json.JSONObject;
+import org.json.JSONArray;
+import java.util.Map;
+import java.util.HashMap;
+
+/**
+ * @version 1.0
+ * @author B. Clint Hall
+ */
+public class Sprites {
+    private Map<String, JSONObject> spriteMap;
+    /**
+     * The root of the sb2 json object is a Stage object.
+     * This constructs a Sprites object from a Stage JSONObject.
+     * @param stage root of an Sb2 file
+     */
+    public Sprites(JSONObject stage) {
+        JSONArray stageChildren = stage.optJSONArray("children");
+        if (stageChildren == null) {
+            stageChildren = new JSONArray();
+        }
+        spriteMap = new HashMap<String, JSONObject>();
+        for (int i = 0; i < stageChildren.length(); i++) {
+            JSONObject child = stageChildren.optJSONObject(i);
+            if (isSprite(child)) {
+                spriteMap.put(child.getString("objName"), child);
+            }
+        }
+    }
+    /**
+     * Check whether a stage child is a Sprite.  It could also be
+     * a StageMonitor.
+     * @param stageChild A JSONObject that may represent a Sprite or a
+     *        StageMonitor
+     * @return whether the stageChild is a Sprite.
+     */
+    private boolean isSprite(JSONObject stageChild) {
+        return stageChild != null && stageChild.has("spriteInfo");
+    }
+    /**
+     * Each sprite has a unique name.
+     * This method returns these names in a array of strings.
+     * @return Array of sprite names.
+     */
+    public String[] getSpriteNames() {
+        String[] spriteNames = new String[spriteMap.size()];
+        return spriteMap.keySet().toArray(spriteNames);
+    }
+
+
+    /**
+     * Each script is represented by an json array of block tuples,
+     * where each block tuple is like a line of code. A Sprite may have
+     * multiple Scripts associated with it.  This function returns an array
+     * of json arrays of block tuples for the requested Sprite.
+     * @param spriteName The name of the sprite
+     * @return An array block tuples.  Each block tuple is represented as
+     *         a JSONArray.
+     */
+    private JSONArray[] getScripts(String spriteName) {
```

Comment:
This method seems to do two things:
- Extract the scripts from the sprite
- Extract individual blocks from scripts
Maybe I'm just not understanding the practical difference between the scripts as a whole and the collection of block tuples that we are building. Is the scripts property of the sprite useless without this extra processing? Should this method be renamed to reflect the fact that it is returning a collection of block tuples rather than a collection of scripts?